src/hooks/useHandleQueryErrors/useHandleQueryErrors.ts

```diff
@@ -0,0 +1,27 @@
+import { isClientError, isServerError } from 'utils/apiErrorStatuses';
+import { ApiError } from 'context/apiClient/apiClientContextController/apiError/apiError.types';
+import { ExtendedQueryMeta } from 'api/types/types';
+
+export const useHandleQueryErrors = () => {
+  const handleErrors = (error: ApiError) => {
+    if (isServerError(error.statusCode)) {
+      // show translated error message in toast/snackbar
+      // handle logger
+    }
+
+    if (isClientError(error.statusCode)) {
+      // show translated error message in toast/snackbar
+      // handle logger
+    }
+  };
+
+  const shouldHandleGlobalError = (metaError?: ExtendedQueryMeta['error'], errorCode?: number) => {
```

Comment:
why is this method here will it be used anywhere other than the provider?

Reply:
TBH I copied this hook from https://github.com/TheSoftwareHouse/react-starter-boilerplate/pull/62. I'm not sure if this whole hook will ever be used somewhere outside of the provider. I think it was made to not bloat the `ApiClientContextController`. Maybe I could just colocate it closer to the ContextController and rename it to `useHandleGlobalQueryErrors` 
Otherwise it would have to be more generic and accept `onServerError` / `onClientError` but then again I'm not sure if it would any value as a generic hook. especially that `onError` is deprecated for Queries (it's available only in global QueryCache) https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose#a-bad-api